internal/cli/push.go

```diff
@@ -0,0 +1,89 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/substratusai/substratus/internal/cli/utils"
+	"github.com/substratusai/substratus/internal/tui"
+)
+
+func pushCommand() *cobra.Command {
+	var flags struct {
+		namespace  string
+		filename   string
+		kubeconfig string
+	}
+
+	run := func(cmd *cobra.Command, args []string) error {
+		defer tui.LogFile.Close()
+
+		kubeconfigNamespace, restConfig, err := utils.BuildConfigFromFlags("", flags.kubeconfig)
+		if err != nil {
+			return fmt.Errorf("rest config: %w", err)
+		}
+
+		clientset, err := kubernetes.NewForConfig(restConfig)
+		if err != nil {
+			return fmt.Errorf("clientset: %w", err)
+		}
+
+		path := "."
+		if len(args) > 0 {
+			path = args[0]
+		}
+
+		tui.P = tea.NewProgram((&tui.PushModel{
+			Ctx:      cmd.Context(),
+			Path:     path,
+			Filename: flags.filename,
+			Namespace: tui.Namespace{
+				Contextual: kubeconfigNamespace,
+				Specified:  flags.namespace,
+			},
+			Client: NewClient(clientset, restConfig),
+			K8s:    clientset,
+		}).New())
+		if _, err := tui.P.Run(); err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	cmd := &cobra.Command{
+		Use:   "push [dir]",
```

Comment:
I'm fine with `push` as well but feel `run` is more descriptive of what the end-result is here. The pushing is just an intermediary step to run it.

Reply:
Updated to `run`